Tribler/Core/Libtorrent/LibtorrentMgr.py

```diff
@@ -631,9 +655,326 @@ def post_session_stats(self, hops=None):
+                    await self.get_download(infohash).stop()
+                    self.tribler_session.notifier.notify(NTFY_TORRENT, NTFY_ERROR, infohash, repr(ds.get_error()), is_hidden)
+            elif state == DLSTATUS_SEEDING:
+                seeding_download_list.append({u'infohash': infohash,
+                                              u'download': download})
+
+                if self.tribler_session.bootstrap and not self.tribler_session.bootstrap.bootstrap_finished \
+                        and hexlify(infohash) == self.tribler_session.config.get_bootstrap_infohash() \
+                        and self.tribler_session.trustchain_community:
+                    if download.future_finished.done():
+                        get_event_loop().run_in_executor(None, self.tribler_session.import_bootstrap_file)
+                        self.tribler_session.bootstrap.bootstrap_finished = True
+                    else:
+                        self._logger.info("Bootstrap download not finished yet, rescheduling")
+
+                if infohash in self.previous_active_downloads:
+                    self.tribler_session.notifier.notify(NTFY_TORRENT, NTFY_FINISHED, infohash, safename, is_hidden)
+                    do_checkpoint = True
+                elif download.config.get_hops() == 0 and download.config.get_safe_seeding():
+                    # Re-add the download with anonymity enabled
+                    hops = self.tribler_session.config.get_default_number_hops()
+                    await self.update_download_hops(download, hops)
+
+            # Check the peers of this download every five seconds and add them to the payout manager when
+            # this peer runs a Tribler instance
+            if self.state_cb_count % 5 == 0 and download.config.get_hops() == 0 and self.tribler_session.payout_manager:
+                for peer in download.get_peerlist():
+                    if str(peer["extended_version"]).startswith('Tribler'):
+                        self.tribler_session.payout_manager.update_peer(unhexlify(peer["id"]), infohash, peer["dtotal"])
+                        if self.tribler_session.bootstrap and hexlify(infohash) == self.tribler_session.config.get_bootstrap_infohash():
+                            if not self.is_pending_task_active('fetch_bootstrap_peers'):
+                                self.register_task('fetch_bootstrap_peers', self.tribler_session.bootstrap.fetch_bootstrap_peers)
+
+        self.previous_active_downloads = new_active_downloads
+        if do_checkpoint:
+            await self.checkpoint_downloads()
+
+        if self.state_cb_count % 4 == 0:
+            if self.tribler_session.tunnel_community:
+                self.tribler_session.tunnel_community.monitor_downloads(states_list)
+            if self.tribler_session.credit_mining_manager:
+                self.tribler_session.credit_mining_manager.monitor_downloads(states_list)
+
+        return []
+
+    # TODO: Make it async?
```

Comment:
This is just adding downloads, which as far as I know is already async (although the function itself is not defined as `async`).

Reply:
Good point, removed notice.